builtin/logical/database/rotation.go

```diff
@@ -530,6 +533,16 @@ func (b *databaseBackend) setStaticAccount(ctx context.Context, s logical.Storag
 	_, err = dbi.database.UpdateUser(ctx, updateReq, false)
 	if err != nil {
 		b.CloseIfShutdown(dbi, err)
+		if usedExistingCredential {
+			// A retried password has failed again. Delete WAL and try with fresh password
```

Comment:
nit: This comment is a little unclear to me. I have context here so I understand what it is trying to say. But I am not so sure about someone reading this that doesn't have the context.

I think if we renamed `usedExistingCredential` to `usedCredentialFromPreviousRotation` then we could omit this comment entirely. If we do that, I think the comment on line 543 should be sufficient to explain the expectations going forward.

Reply:
That makes sense! Renamed in latest commit